src/cache.rs

```diff
@@ -100,124 +138,167 @@ pub fn id() -> usize {
 /// Comemo's cache is thread-local, meaning that this only evicts this thread's
 /// cache.
 pub fn evict(max_age: usize) {
-    CACHE.with(|cache| {
-        let mut cache = cache.borrow_mut();
-        cache.map.retain(|_, entries| {
-            entries.retain_mut(|entry| {
-                entry.age += 1;
-                entry.age <= max_age
-            });
-            !entries.is_empty()
-        });
-    });
-    ACCELERATOR.with(|accelerator| accelerator.borrow_mut().clear());
+    CACHES.read().iter().for_each(|fun| fun(max_age));
```

Comment:
I think a for loop is more readable. `fun` could also be named more respectively, e.g. `subevict`.